src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -34,6 +34,7 @@ public partial class NatsConnection : INatsConnection
     /// Hook before TCP connection open.
     /// </summary>
     public Func<(string Host, int Port), ValueTask<(string Host, int Port)>>? OnConnectingAsync;
+    public Func<ISocketConnection, Task>? OnSocketAvailableAsync;
```

Comment:
we should pull this up to `INatsConnection` as well. also can we return `ValueTask` to be consistent?